### ISSUE TITLE
Terminals such as `kitty` and `alacritty` suddenly take ~1.5s to display a window

### ISSUE THREAD

**Issue by expenses** [swaywm/sway#7124]
I'm not 100% sure this is a sway problem, but I'll post here anyway.

### Please fill out the following:
- **Sway Version:** `sway version 1.7`

- **Debug Log:** [sway_debug.log.zip](https://github.com/swaywm/sway/files/9229391/sway_debug.log.zip)

- **Description:**

Over the last couple of days I have noticed that `alacritty` had gone from starting up instantaneously to taking roughly 1.5 seconds to startup (See the below log). I tested `kitty` and had the same problem. `foot`, `firefox` and `chromium` are still instantaneous to display a window though. I'm running Arch Linux (x86_64) with the 5.18.14-arch1-1 kernel. I suspect this is either a sway/wayland problem or a graphics API/driver one (although `glxgears` and `vulkancube` are also instantaneous).

Edit: just tested i3 on X11 and `alacritty` and `kitty` both worked fine there.

`alacritty -vvv`:

```
Created log file at "/tmp/Alacritty-12769.log"
[0.000006651s] [INFO ] [alacritty] Welcome to Alacritty
[0.000056448s] [INFO ] [alacritty] Version 0.10.1 ()
[0.000073880s] [INFO ] [alacritty] Running on Wayland
[0.000247013s] [INFO ] [alacritty_config_derive] No config file found; using default
[0.012559714s] [TRACE] [crossfont] Number of fonts is 30
[0.012586742s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Regular.otf", index=0
[0.012654944s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.013029001s] [DEBUG] [alacritty] Estimated DPR: 1
[0.013036514s] [DEBUG] [alacritty] Estimated window size: None
[0.013039806s] [DEBUG] [alacritty] Estimated cell size: 9 x 18
[1.403650472s] [INFO ] [alacritty] Device pixel ratio: 1
[1.409021168s] [INFO ] [alacritty] Initializing glyph cache...
[1.409405705s] [TRACE] [crossfont] Number of fonts is 30
[1.409421831s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Regular.otf", index=0
[1.409474838s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.410056006s] [TRACE] [crossfont] Number of fonts is 51
[1.410067379s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Bold.otf", index=0
[1.410114960s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.410838179s] [TRACE] [crossfont] Number of fonts is 48
[1.410849410s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Italic.otf", index=0
[1.410884377s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.411652491s] [TRACE] [crossfont] Number of fonts is 67
[1.411668096s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-BoldItalic.otf", index=0
[1.411706176s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.418193280s] [INFO ] [alacritty] ... finished initializing glyph cache in 0.009157762s
[1.418206644s] [INFO ] [alacritty] Cell size: 9 x 18
[1.418209546s] [INFO ] [alacritty] Padding: 0 x 0
[1.418211382s] [INFO ] [alacritty] Width: 800, Height: 600
[1.418755407s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 197, g: 200, b: 198 } to Rgb { r: 130, g: 132, b: 130 }
[1.418770637s] [TRACE] [alacritty] Deriving dim colors from normal colors
[1.418773093s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 29, g: 31, b: 33 } to Rgb { r: 19, g: 20, b: 21 }
[1.418775320s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 204, g: 102, b: 102 } to Rgb { r: 134, g: 67, b: 67 }
[1.418777234s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 181, g: 189, b: 104 } to Rgb { r: 119, g: 124, b: 68 }
[1.418778984s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 240, g: 198, b: 116 } to Rgb { r: 158, g: 130, b: 76 }
[1.418780746s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 129, g: 162, b: 190 } to Rgb { r: 85, g: 106, b: 125 }
[1.418782509s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 178, g: 148, b: 187 } to Rgb { r: 117, g: 97, b: 123 }
[1.418784317s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 138, g: 190, b: 183 } to Rgb { r: 91, g: 125, b: 120 }
[1.418785975s] [TRACE] [alacritty_terminal] Scaling RGB by 0.66 from Rgb { r: 197, g: 200, b: 198 } to Rgb { r: 130, g: 132, b: 130 }
[1.418793641s] [INFO ] [alacritty] PTY dimensions: 33 x 88
[1.420394591s] [INFO ] [alacritty] Initialisation complete
[1.422892061s] [DEBUG] [alacritty_terminal] New num_cols is 83 and num_lines is 52
[1.424657841s] [INFO ] [alacritty] Padding: 0 x 0
[1.424671454s] [INFO ] [alacritty] Width: 748, Height: 947
[1.425520725s] [TRACE] [alacritty_terminal] Carriage return
[1.425582225s] [TRACE] [alacritty_terminal] Linefeed
[1.425626301s] [TRACE] [alacritty_terminal] Setting attribute: Foreground(Named(Green))
[1.425670172s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.425685721s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.425696705s] [TRACE] [alacritty_terminal] Carriage return
[1.425701784s] [TRACE] [alacritty_terminal] Linefeed
[1.428081360s] [TRACE] [alacritty_terminal] Setting mode: BracketedPaste
[1.430952256s] [TRACE] [crossfont] Number of fonts is 30
[1.430978534s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Regular.otf", index=0
[1.430989907s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.431630900s] [TRACE] [crossfont] Number of fonts is 51
[1.431645327s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Bold.otf", index=0
[1.431655122s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.432370395s] [TRACE] [crossfont] Number of fonts is 48
[1.432384924s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-Italic.otf", index=0
[1.432390541s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.433063648s] [TRACE] [crossfont] Number of fonts is 67
[1.433075889s] [TRACE] [crossfont] Got font path="/usr/share/fonts/gsfonts/NimbusMonoPS-BoldItalic.otf", index=0
[1.433081397s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[1.433587044s] [INFO ] [alacritty] Font size changed to Size(22) with DPR of 2
[1.438255071s] [INFO ] [alacritty] Cell size: 18 x 35
[1.438265488s] [DEBUG] [alacritty_terminal] New num_cols is 83 and num_lines is 54
[1.439284578s] [INFO ] [alacritty] Padding: 0 x 0
[1.439294308s] [INFO ] [alacritty] Width: 1496, Height: 1894
[1.439303602s] [TRACE] [alacritty_terminal] Setting title to 'Some("~")'
[1.439322576s] [TRACE] [alacritty_terminal] Setting attribute: Foreground(Named(Black))
[1.439329078s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.439337333s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.439341214s] [TRACE] [alacritty_terminal] Carriage return
[1.439350541s] [TRACE] [alacritty_terminal] Setting attribute: Foreground(Named(BrightGreen))
[1.439354443s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.439358073s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.439361240s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.439363517s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.439365446s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.439367263s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.439369225s] [TRACE] [alacritty_terminal] Setting attribute: Foreground(Named(Green))
[1.439371516s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.439373812s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.439375894s] [TRACE] [alacritty_terminal] Configuring charset G0 as Ascii
[1.439377909s] [TRACE] [alacritty_terminal] Setting attribute: Reset
[1.439380563s] [TRACE] [alacritty_terminal] Clearing line: Right
[1.439383139s] [TRACE] [alacritty_terminal] Carriage return
[1.439385624s] [TRACE] [alacritty_terminal] Moving forward: 20
```

**Comment by bl4ckb0ne** [comment]
I don't see anything weird. Could you run either one of them with `WAYLAND_DEBUG=1` please.

**Comment by nullndr** [comment]
Same here for alacritty.

In my case is like 2~3 seconds.

Even in my case other applications like firefox open instantaneous.

**Comment by bl4ckb0ne** [comment]
Nothing seems weird, I'm not able to reproduce on my end with alacritty or kitty

```
$ sway --version
sway version 1.7
$ alacritty --version
alacritty 0.10.1 (1370e8d5bf)
$ kitty --version
kitty 0.25.2 created by Kovid Goyal
```

Also

```
[0.013301030s] [DEBUG] [alacritty] Estimated cell size: 9 x 21
[0.056827059s] [INFO ] [alacritty] Device pixel ratio: 1
```

compared to the log from the initial post

```
[0.013039806s] [DEBUG] [alacritty] Estimated cell size: 9 x 18
[1.403650472s] [INFO ] [alacritty] Device pixel ratio: 1
```

**Comment by nullndr** [comment]
## Update

I updated my system and now when I am on a new workspace alacritty open instantaneous.

This does not happen when there are another application like Firefox or VSCode open, but I use the `Alt` key as `$mod`, could it be a conflict?

**Comment by bl4ckb0ne** [comment]
@nullndr the problem occurs again if you have firefox or vscode open in the same workspace?

**Comment by nullndr** [comment]
yup

**Comment by nullndr** [comment]
> I use the Alt key as $mod, could it be a conflict?

I tested other applications like Discord that do not use the `Alt` key and the problem did not occur, so could it really be a conflict?

**Comment by bl4ckb0ne** [comment]
Might be, you'll have to try another `$mod` to see if it occurs again. I use `Mod4` and opening alacritty/kitty with firefox in the same workspace is instantaneous.

**Comment by nullndr** [comment]
Yeah, it works

**Comment by bl4ckb0ne** [comment]
Waiting for confirmation by @expenses before closing

**Comment by nullndr** [comment]
Isn't there something that sway could do?

**Comment by bl4ckb0ne** [comment]
There's not much sway can do if several applications are using the same meta key.

**Comment by nullndr** [comment]
I see, thank you for your time

**Comment by expenses** [comment]
> Waiting for confirmation by @expenses before closing

I also updated, and now the terminals are launching instantaneously again

**Comment by expenses** [comment]
I know that this issue is closed but I want to link to #7600 as it contains the actual fix.
